src/variants.php

```diff
@@ -69,36 +69,46 @@
 
 
 
-if (!ACTION && (empty($_PE[1]) || preg_match('/^chr[0-9A-Z]{1,2}$/', $_PE[1]))) {
+if (!ACTION && (empty($_PE[1]) ||
+    preg_match('/^(chr[0-9A-Z]{1,2})(:([0-9,]+)-([0-9,]+))?$/', $_PE[1], $aRegionArgs))) {
     // URL: /variants
     // URL: /variants/chrX
+    // URL: /variants/chr3:20-200,000
     // View all genomic variant entries, optionally restricted by chromosome.
 
     // Managers are allowed to download this list...
     if ($_AUTH['level'] >= LEVEL_MANAGER) {
         define('FORMAT_ALLOW_TEXTPLAIN', true);
     }
 
-    if (!empty($_PE[1])) {
-        $sChr = $_PE[1];
-    } else {
-        $sChr = '';
+    require_once ROOT_PATH . 'class/object_genome_variants.php';
+    $_DATA = new LOVD_GenomeVariant();
+    $aColsToHide = array('allele_');
+    $sTitle = 'View all genomic variants';
+
+    // Set conditions on viewlist if a region is specified (e.g. chr3:20-200,000)
+    if (isset($aRegionArgs)) {
+        // Set search condition for chromosome.
+        $_GET['search_chromosome'] = '="' . substr($aRegionArgs[1], 3) . '"';
+        $aColsToHide[] = 'chromosome';
+
+        if (count($aRegionArgs) == 5) {
+            // Set search conditions for start and end of region.
+            $_GET['search_position_g_start'] = '>' . str_replace(',', '', $aRegionArgs[3]);
+            $_GET['search_position_g_end'] = '<' . str_replace(',', '', $aRegionArgs[4]);
+            $sTitle .= ' in region ' . $aRegionArgs[0];
+        } else {
+            $sTitle .= ' at chromosome ' . substr($aRegionArgs[1], 3);
+        }
```

Comment:
You could make this code a bit more readable by running a `list()` on `$aRegionArgs`, so that the values are clear. Anyone can read back of course, but it's much nicer to have a `$sChr`, `$nPositionStart`, and `$nPositionEnd` (for instance). You can then also get rid of the combined position fields by telling `preg_match()` not to count that subpattern using `?:`, like so:
```
preg_match('/^(chr[0-9A-Z]{1,2})(?::([0-9,]+)-([0-9,]+))?$/', $_PE[1], $aRegionArgs)
```
(note that then you'll need to change the check on the `count()` to 4 then)